### ISSUE TITLE
PR: Roofs with inserts not exported as aggregations in IFC4 RVArch

### ISSUE THREAD

**Issue by andydandy74** [Autodesk/revit-ifc#652]
### Problem Description

When a roof contains an inserted object (e.g. an opening) it is exported as a singular ```IfcRoof``` element, not as an aggregation that contains an ```IfcSlab``` element. Since this didn't happen in IFC2x3 CV2 (and still doesn't) I am curious whether this is by design or should be treated as a bug? It would be great to have the same behaviour (export as aggregation) here in both cases - e.g. the nested ```IfcSlab``` provides better base quantities.

![grafik](https://github.com/Autodesk/revit-ifc/assets/3014437/e96eb377-1889-4b5d-ae23-25c5cd4ec4e4)

### Revit Version

2023.1.x

### IFC for Revit Addon Version

23.2.5

### Windows Version

10 22H2

**Comment by AngelVelezSosa** [comment]
It's both a possible improvement (vs a bug) and a funny think you should mention...
1.  Possible improvement: I agree that we should export aggregates when we can.  Not exporting an aggregate is allowed, though, so isn't technically a bug.  But we could try to detect these cases better.
2. Some customers are complaining when it is exported as an aggregate at all, so we have looked at the possibility of allowing a checkbox to never export as an aggregate.  I wouldn't use that checkbox, but others would.

**Comment by andydandy74** [comment]
Yes, I'm aware it's allowed to not export an aggregate and I wasn't entirely sure whether to post it as an issue, a feature request or a question. :-)

**Comment by ArtemHnatko** [comment]
This issue is also reported in Jira as "REVIT-235642" and is currently in the "Open" status.
